src/search/dag_classic/search.cc

```diff
@@ -2041,67 +2045,43 @@ void SearchWorker::RunNNComputation() {
-    auto wdl_rescale = [&]() {
-      if (params_.GetWDLRescaleRatio() != 1.0f ||
-          (params_.GetWDLRescaleDiff() != 0.0f &&
-           search_->contempt_mode_ != ContemptMode::NONE)) {
-        // Check whether root moves are from the set perspective.
-        bool root_stm = search_->contempt_mode_ == ContemptMode::WHITE;
-        auto sign = (root_stm ^ node_to_process->history.IsBlackToMove())
-                        ? 1.0f
-                        : -1.0f;
-        WDLRescale(node_to_process->eval->q, node_to_process->eval->d,
-                   params_.GetWDLRescaleRatio(),
-                   search_->contempt_mode_ == ContemptMode::NONE
-                       ? 0
-                       : params_.GetWDLRescaleDiff(),
-                   sign, false, params_.GetWDLMaxS());
-      }
-    };
-    if (is_tt_miss) {
-      assert(!tt_iter->second.expired());
-      wdl_rescale();
-      node_to_process->tt_low_node->SetNNEval(node_to_process->eval.get());
-      node_to_process->tt_low_node->SortEdges();
-    } else {
-      auto tt_low_node = tt_iter->second.lock();
-      if (!tt_low_node) {
-        tt_iter->second = node_to_process->tt_low_node;
-        wdl_rescale();
-        node_to_process->tt_low_node->SetNNEval(node_to_process->eval.get());
-        node_to_process->tt_low_node->SortEdges();
-      } else {
-        assert(!tt_iter->second.expired());
-        node_to_process->tt_low_node = tt_iter->second.lock();
-      }
+  auto wdl_rescale = [&]() {
+    if (params_.GetWDLRescaleRatio() != 1.0f ||
+        (params_.GetWDLRescaleDiff() != 0.0f &&
+         search_->contempt_mode_ != ContemptMode::NONE)) {
+      // Check whether root moves are from the set perspective.
+      bool root_stm = search_->contempt_mode_ == ContemptMode::WHITE;
+      auto sign = (root_stm ^ node_to_process->history.IsBlackToMove())
+                      ? 1.0f
+                      : -1.0f;
+      WDLRescale(node_to_process->eval->q, node_to_process->eval->d,
+                 params_.GetWDLRescaleRatio(),
+                 search_->contempt_mode_ == ContemptMode::NONE
+                     ? 0
+                     : params_.GetWDLRescaleDiff(),
+                 sign, false, params_.GetWDLMaxS());
     }
-  }
+  };
+  wdl_rescale();
+  node_to_process->tt_low_node->SetNNEval(node_to_process->eval.get());
+  node_to_process->tt_low_node->SortEdges();
 
   // Add NN results to node.
   Node* node = node_to_process->node;
   // Add Dirichlet noise if enabled and at root.
   if (params_.GetNoiseEpsilon() && node == search_->root_node_) {
-    node->SetLowNode(
-        std::make_shared<LowNode>(*node_to_process->tt_low_node.get()));
-    ApplyDirichletNoise(node, params_.GetNoiseEpsilon(),
-                        params_.GetNoiseAlpha());
-    node->SortEdges();
-  } else {
-    node->SetLowNode(node_to_process->tt_low_node);
+    ApplyDirichletNoise(node_to_process->tt_low_node.get(),
```

Comment:
We only use Dirichlet noise for training data generation, which we don't do with dag. So if it makes things easier, feel free to remove it.